src/raft/raft.rs

```diff
@@ -1624,7 +1633,7 @@ impl<T: Storage> Raft<T> {
 
         // The quorum size is now smaller, so see if any pending entries can
         // be committed.
-        if self.maybe_commit() {
+        if self.maybe_commit() && !self.skip_bcast_commit {
```

Comment:
No, commit index should be updated no matter we want to bcast it or not, which will make all following heartbeats and append messages contain the latest commit index.